src/Build/Graph/ProjectGraph.cs

```diff
@@ -250,6 +252,84 @@ private void LoadGraph(string entryProjectFile, ProjectCollection projectCollect
             }
         }
 
+        // Using Tarjan's algorithm for detecting strongly connected components in a directed graph
+        // Reference: https://en.wikipedia.org/wiki/Strongly_connected_component
+        private int _index = 0;
```

Comment:
Does this need to be a field on the class? Could it just be passed around as a param?

If not, rename it to something less generic (`_cycleDetectionDepth`?)